src/lib.rs

```diff
@@ -194,6 +207,22 @@ fn handle_error(app: &Config, req: &mut Request, err: BrokerError) -> IronResult
 }
 
 
+/// Invoked after handlers for all responses, to set common headers.
+fn set_default_headers(res: &mut Response) {
+    // Specify a tight content security policy. We need to be able to POST
+    // redirect anywhere, and run our own inline scripts.
+    let csp = vec![
+        "sandbox allow-scripts allow-forms",
+        "default-src 'none'",
+        "script-src 'unsafe-inline'",
```

Comment:
We should add a bug to further lock down script-src. Which would mean a followup PR that moves our JavaScript out into its own files.